base/system/userinit/userinit.c

```diff
@@ -258,17 +255,33 @@ TryToStartShell(
                         &si,
                         &pi))
     {
-        WARN("CreateProcess() failed with error %lu\n", GetLastError());
+        WARN("CreateProcessW() failed with error %lu\n", GetLastError());
         return FALSE;
     }
 
-    StartAutoApplications(CSIDL_STARTUP);
-    StartAutoApplications(CSIDL_COMMON_STARTUP);
     CloseHandle(pi.hProcess);
     CloseHandle(pi.hThread);
     return TRUE;
 }
 
+static BOOL
+TryToStartShell(
+    IN LPCWSTR Shell)
+{
+    WCHAR ExpandedShell[MAX_PATH];
+
+    TRACE("(%s)\n", debugstr_w(Shell));
+
+    ExpandEnvironmentStringsW(Shell, ExpandedShell, ARRAYSIZE(ExpandedShell));
```

Comment:
Can we end up with a path longer than MAX_PATH ?
If so, need to check if it failed.